### PR TITLE
[Rollup] Validate timezone in range queries

### PR DESCRIPTION
**Pull request by polyfractal**
When validating the search request, we make sure any date_histogram aggregations have timezones that match the jobs.  But we don't do similar validation on range queries.

While it wouldn't produce incorrect results, it would be confusing to the user as no documents would match the aggregation (because we add a filter clause on the timezone for the agg).

Now the user gets an exception up front, some helpful text about why the range query didn't match, and which timezones are acceptable

### CODE REVIEW

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/action/SearchActionTests.java [hunk 1]
```diff
@@ -114,6 +114,24 @@ public void testBadQuery() {
...
+        GroupConfig.Builder group = ConfigTestHelpers.getGroupConfig();
+        group.setDateHisto(new DateHistoGroupConfig.Builder().setField("foo").setInterval(new DateHistogramInterval("1h")).build());
+        job.setGroupConfig(group.build());
+        RollupJobCaps cap = new RollupJobCaps(job.build());
+        Set<RollupJobCaps> caps = new HashSet<>();
+        caps.add(cap);
+        QueryBuilder rewritten = null;
+        try {
+            rewritten = TransportRollupSearchAction.rewriteQuery(new RangeQueryBuilder("foo").gt(1).timeZone("UTC"), caps);
+        } catch (Exception e) {
```
Comment (jimczi) [inline x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/action/SearchActionTests.java]: Maybe not needed to catch the exception here and just make testRange throws Exception ?

**Review by jimczi — APPROVED** [review]
Left a minor comment, LGTM otherwise

File: x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/action/SearchActionTests.java (hunk 1 above)
Comment (polyfractal) [reply x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/action/SearchActionTests.java]: :+1:

Comment (polyfractal) [reply x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/action/SearchActionTests.java]: Looks like there are several of these, will clean them all up

**Pull request merged**